File: Day 6/Sum_of_2/app/src/main/java/com/example/sum_of_2/MainActivity.java
package com.example.sum_of_2;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private EditText editTextText2;

    private EditText editTextText3;

    private TextView textView2 ;

    private Button button;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        editTextText2 = findViewById(R.id.editTextText2);
        editTextText3 = findViewById(R.id.editTextText3);
        textView2 = findViewById(R.id.textView2);
        button = findViewById(R.id.button);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                String a = editTextText2.getText().toString();
                String b = editTextText3.getText().toString();

                int m = Integer.parseInt(a);
                int n = Integer.parseInt(b);

                int c = m+n;

                textView2.setText("the Sum of 2 number is " + c );



            }
        });





    }

//    public void sum(View view)
//    {
//
//        String a = editTextText2.getText().toString();
//        String b = editTextText3.getText().toString();
//
//        int m = Integer.parseInt(a);
//        int n = Integer.parseInt(b);
//
//        int c = m+n;
//
//        textView2.setText("the Sum of 2 number is " + c );
//
//
//
//
//
//
//
//    }
}

File: Siddhi mam Tasks/Day 37/Co-routines/Coroutine/app/src/main/java/com/example/coroutine/MainActivity.kt
package com.example.coroutine

import android.content.ContentValues.TAG
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        CoroutineScope(Dispatchers.IO).launch{
            printFollowers()
        }
    }

    private suspend fun printFollowers(){
        var fbfollowers = 0
        val job = CoroutineScope(Dispatchers.IO).launch {
            fbfollowers=getFollowers()
        }

        job.join()       // if we cmt this line then as we know it is co-routine then it will look for the other task to complete and it will print the 0 because getfollower is delay by 1000 ms that why it will directly print 0
                         // join method will keep the object in the waiting state so as long as we will not get the return value it will not go for other task
        Log.d(TAG,fbfollowers.toString())
    }

    private suspend fun getFollowers(): Int {

        delay(1000)
        return 54

    }
}